### PR TITLE
Add default keyword handling in rustdoc

### LINKED ISSUE #58898: Rustdoc ommits `default` for `default fn`
screenshot_2019-03-03 rustc ty query values value - rust

```rust
impl<'tcx, T> Value<'tcx> for T {
    default fn from_cycle_error<'a>(tcx: TyCtxt<'a, 'tcx, 'tcx>) -> T {
        tcx.sess.abort_if_errors();
        bug!("Value::from_cycle_error called without errors");
    }
}
```

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #58898.

r? @QuietMisdreavus

### CODE REVIEW

File: src/test/rustdoc/default_trait_method.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#![feature(specialization)]
+
+pub trait Item {
+    fn foo();
+    fn bar();
+}
+
+// @has default_trait_method/trait.Item.html
+// @has - '//*[@id="method.foo"]' 'default fn foo()'
+// @has - '//*[@id="method.bar"]' 'fn bar()'
```
Comment (QuietMisdreavus) [inline src/test/rustdoc/default_trait_method.rs]: Is there a way we can make sure that `bar()` does not have the `default` keyword? Would `@count - 'default' 1` work for this test?

Comment (GuillaumeGomez) [reply src/test/rustdoc/default_trait_method.rs]: Good question. I'll check.

Comment (GuillaumeGomez) [reply src/test/rustdoc/default_trait_method.rs]: `@count` is used to count DOM elements, not strings occurrence that won't work. I strengthen the test.

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by QuietMisdreavus — APPROVED** [review]

**Pull request merged**